### PR TITLE
Upload 08 15 24

### PR DESCRIPTION
**Pull request by bishopkws**

### CODE REVIEW

File: data/indicator_2-2-1.csv [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-2000,3.2
-2001,3.2
-2003,3.1
-2005,2.9
-2007,3.1
-2009,2.3
-2011,2.8
-2013,2.2
-2015,2.4
+﻿Year,Corrected_Value
```
Comment (btylerburton) [inline data/indicator_2-2-1.csv:1]: This needs to be `Value` and not `Corrected_Value`

**Review by btylerburton — CHANGES_REQUESTED** [review]
All files need to be encoded in `UTF-8`, not `UTF-8 with BOM`

```
 File "scripts/check_data.py", line 6, in <module>
    validation_successful = open_sdg_check(config='config_data.yml', alter_meta=alter_meta, alter_data=alter_data)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/sdg/open_sdg.py", line 306, in open_sdg_check
    outputs = open_sdg_prep(options)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/sdg/open_sdg.py", line 359, in open_sdg_prep
    ignore_out_of_scope_disaggregation_stats=options['ignore_out_of_scope_disaggregation_stats'],
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/sdg/outputs/OutputOpenSdg.py", line 36, in __init__
    indicator_options, logging=logging)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/sdg/outputs/OutputBase.py", line 42, in __init__
    self.indicators = self.merge_inputs(inputs)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/sdg/outputs/OutputBase.py", line 121, in merge_inputs
    input.execute_once(self.indicator_options)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/sdg/inputs/InputBase.py", line 51, in execute_once
    self.execute(indicator_options)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/sdg/inputs/InputCsvData.py", line 32, in execute
    data = pd.read_csv(indicator_map[inid], dtype=self.dtype)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 586, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 482, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 811, in __init__
    self._engine = self._make_engine(self.engine)
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/pandas/io/parsers/readers.py", line 1040, in _make_engine
    return mapping[engine](self.f, **self.options)  # type: ignore[call-arg]
  File "/Users/brendantburton/.pyenv/versions/sdg/lib/python3.7/site-packages/pandas/io/parsers/c_parser_wrapper.py", line 69, in __init__
    self._reader = parsers.TextReader(self.handles.handle, **kwds)
  File "pandas/_libs/parsers.pyx", line 542, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 642, in pandas._libs.parsers.TextReader._get_header
  File "pandas/_libs/parsers.pyx", line 843, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas/_libs/parsers.pyx", line 1917, in pandas._libs.parsers.raise_parser_error
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 185: invalid start byte
```

**Comment by bishopkws** [comment]
Thank you. I thought I’d fixed this.

**Comment by bishopkws** [comment]
I’m not quite sure what this means, but I’ll work to figure it out.

**Comment by btylerburton** [comment]
If you use an application like VS Code you can reencode each file manually by clicking on:

1. `UTF-8 with BOM`
2. `Save with encoding`
3. `UTF-8`

This is outlined in the screenshots below. 

If you're having issues, I'd be glad to jump on a call. Speak with JeongSoo to get that setup.


utf-with-bom
Screenshot 2024-08-15 at 12 00 28 PM
Screenshot 2024-08-15 at 12 01 04 PM

**Comment by btylerburton** [comment]
Also, you can use a tool like Notepad+

https://stackoverflow.com/questions/33510700/excel-files-with-byte-order-mark-for-utf-8-causing-errors

**Comment by bishopkws** [comment]
Would you recommend I try to troubleshoot this with the yml files before
trying another batch data pull? Or can I just resubmit the csv files?

Kristi

**Comment by btylerburton** [comment]
You can just resubmit the current files if that's easier. If you're using a process to generate the CSV's you just want to make sure that process doesn't change them yet again after this.

**Comment by brockfanning** [comment]
@bishopkws I'm wondering if maybe it is happening when the data provider exports the file from Excel to CSV. Maybe there is a particular export option that is problematic. If it helps to know a specific indicator, see 3.4.1. You can visually see the problem in the "Files changed" section:

image

This might be a tricky one to track down, but maybe a place to start is a screenshare with the data provider to walk through the process of exporting from Excel to CSV.

**Pull request closed without merging**